common/djangoapps/student/roles.py

```diff
@@ -19,6 +19,13 @@
 # A list of registered access roles.
 REGISTERED_ACCESS_ROLES = {}
 
+# A mapping of roles to the roles that they inherit permissions from.
+# For example, if you want to create a new role that has the same permissions as staff, you would
+# add an entry to this dictionary like: 'staff': ['new_role']
+ACCESS_ROLES_INHERITANCE = {
+    'staff': ['limited_staff'],
```

Comment:
Instead of hardcoding it here, let's do the following:
1. Add `PARENT_ROLE = CourseStaffRole.ROLE` to the `CourseLimitedStaffRole`. (Feel free to change the `PARENT_ROLE` to something more accurate.)
2. Add this to `register_access_role`:
    ```python
        if parent_role := getattr(cls, "PARENT_ROLE", None):
            ACCESS_ROLES_INHERITANCE.setdefault(parent_role, set()).add(cls.ROLE)
    ```
3. Change the `return` statement of the `_get_roles` to `return ACCESS_ROLES_INHERITANCE.get(role, set()) | {role}`.